rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/external/source/ExternalWordSource.kt

```diff
@@ -1,39 +1,25 @@
 package systems.ajax.englishstudytelegrambot.external.source
 
-import org.slf4j.LoggerFactory
-import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
-import org.springframework.web.server.ResponseStatusException
-import systems.ajax.englishstudytelegrambot.dto.GettingPartOfAdditionalInfoAboutWord
+import reactor.core.publisher.Mono
+import systems.ajax.englishstudytelegrambot.dto.external.api.GettingPartOfAdditionalInfoAboutWord
 import systems.ajax.englishstudytelegrambot.property.WordnikProperties
 
 @Component
 class ExternalWordSource(val wordnikProperties: WordnikProperties, val webClient: WebClient) {
 
-    final inline fun <reified T : GettingPartOfAdditionalInfoAboutWord> customGetInfoAboutWord(
+    final inline fun <reified T : GettingPartOfAdditionalInfoAboutWord> customGetInfoAboutWordFromWordnikAPI(
         wordSpelling: String,
         link: String
-    ): String {
-        return runCatching {
-            webClient.get()
-                .uri(
-                    link, wordSpelling, wordnikProperties.tokenKey
-                )
-                .retrieve()
-                .onStatus({ responseStatus ->
-                    responseStatus == HttpStatus.NOT_FOUND
-                }) { throw ResponseStatusException(HttpStatus.NOT_FOUND) }
-                .bodyToFlux(T::class.java)
-                .blockFirst()!!
-                .partOfAdditionalInfoAboutWord
-        }.getOrElse {
-            log.info("additional info for word {} isn't found", wordSpelling)
-            "Missing"
-        }
-    }
-
-    companion object {
-        val log = LoggerFactory.getLogger(this::class.java)
-    }
+    ): Mono<String> =
+        webClient.get()
+            .uri(
+                link, wordSpelling, wordnikProperties.tokenKey
+            )
+            .retrieve()
+            .bodyToFlux(T::class.java)
+            .next()
```

Comment:
First you convert the body to flux, then you just take one item.

Maybe it's better to use `.toEntity<T>()` instead

Reply:
I depend on response from wordnik api. For every request (audio, pronunciation..) it returns array of answers, but i want save only first. So when i try bodyToMono or toEntity it throws 
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `systems.ajax.englishstudytelegrambot.dto.external.api.PronunciationOfWordResponse` from Array value (token `JsonToken.START_ARRAY`)
i dont need to work with  whole Array so I came to this solution